### PR TITLE
new: Refactor and add unit testing to output-related modules

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This pull request refactors and adds unit tests for all output and response-related logic and classes.

## ✔️ How to Test

```
make test
```

### CODE REVIEW

**Comment by lgarber-akamai** [comment]
I'd like some feedback as to whether we should refactor the `OutputHandler.print(...)` function to return the data to be output rather than printing it directly. I'm leaning towards leaving it the same since `OutputHandler` implies it also handles outputting/printing data.

File: linodecli/output.py [hunk 1]
```diff
@@ -173,29 +167,48 @@ def _select_json_elements(self, keys, json_res):
...
+            header=None,
+            value_transform=lambda attr, model: model):
+        """
+        Returns the `content` to be displayed by the corresponding output function.
+        `value_transform` allows functions to specify how each value should be formatted.
+        """
+
         content = []
 
+        if self.headers and header is not None:
```
Comment (zliang-akamai) [inline linodecli/output.py:204]: I think `self.headers` is almost equivalent to `self.headers and header is not None` here unless the default behavior is overridden by the magic method. Is that the case?

**Comment by zliang-akamai** [comment]
Sorry, that was an incorrect comment, deleted.

**Review by zliang-akamai — APPROVED** [review]

**Review by jriddle-linode — APPROVED** [review]
Maybe down the line we can make this a return and print class but I'm not sure how A. that would impact plugins if they use output handler and B. from my perspective it just makes unit testing stuff easier but we already have unit tests so

**Pull request merged**